dpe/src/commands.rs

```diff
@@ -176,6 +177,54 @@ impl TryFrom<&[u8]> for RotateCtxCmd {
     }
 }
 
+#[repr(C)]
```

Comment:
Lol, I was actually independently writing the following in another comment when I received the notification for your comment:

---

It looks to me that the "parsing" logic for commands may be somehow factorized. Here are some alternatives:
1. Use `bytemuck::try_from_bytes()` and `#[derive(bytemuck::AnyBitPattern)]`. This means adding a dependency.
2. Do the same as `bytemuck` but manually (essentially an unsafe cast). This means using unsafe and losing the static checks from the derive macro.
3. Use `bytes::Buf::copy_to_slice()` and `bytes::Buf::get_u32_le()`. This means adding a dependency and still duplicating the struct layout in the parsing code.
4. Do the same as `bytes` but manually (essentially a wrapper around the slice with appropriate reading methods). This means a bit of code, but would remove a lot of error-prone boilerplate.

I think (1) is the best solution, but if adding a dependency is an issue, then probably (4) is better than (2), just because it avoids unsafe. This is also closer to the current code, just factorizing the index arithmetic.

---

I agree with you that (1) and (2) have another advantage over (3) and (4) that I forgot. Not only do they remove duplication between struct layout and parsing logic, but they also don't allocate.

Reply:
Going to leave it out of this PR, but I agree what we're doing today is a bit cumbersome and error-prone. I opened #40 to track this.